### PR TITLE
Update dependencies and fix code formatting

### LINKED ISSUE #12: CPv3 - Upgrade from Cairo 2.3 to 2.5 (2.6 is release)
**Description**
---
Upgrade the Cairo version to stay up to date and access new feature of Cairo for the Protocol

**Acceptance Criteria**
---

- [ ] Upgrade Cairo 2.3 to 2.5/2.6
- [ ] Implement new features (ByteArray, Public exposition, etc....)

### PR DESCRIPTION
**Pull request by cloudvenger**
Fix #12

### CODE REVIEW

File: src/components/absorber/carbon.cairo [hunk 1]
```diff
@@ -75,10 +73,10 @@ mod AbsorberComponent {
             times[times.len() - 1]
         }
         fn get_times(self: @ComponentState<TContractState>) -> Span<u64> {
-            self.Absorber_times.read().array().span()
+            self.Absorber_times.read().array().unwrap_or_default().span()
```
Comment (tekkac) [inline src/components/absorber/carbon.cairo]:
```suggestion
            self.Absorber_times.read().array().expect('Can\'t get times').span()
```
We might need to handle the case of error instead of having an empty array.

File: src/contracts/project.cairo [hunk 2]
```diff
@@ -102,56 +102,55 @@ mod Project {
...
         self.src5.register_interface(IERC165_BACKWARD_COMPATIBLE_ID);
     }
 
     // Externals
-    #[external(v0)]
+    #[abi(embed_v0)]
     impl ExternalImpl of super::IExternal<ContractState> {
         fn mint(ref self: ContractState, to: ContractAddress, token_id: u256, value: u256) {
-            self.erc1155._mint(to, token_id, value);
+            self.erc1155.mint_with_acceptance_check(to, token_id, value, array![].span());
```
Comment (tekkac) [inline src/contracts/project.cairo:117]: Why the change?
IIRC, we had some issues with safe_transfer with wallets.

**Review by tekkac — CHANGES_REQUESTED** [review]
Minor changes requested, otw lgtm!

File: src/contracts/project.cairo (hunk 2 above)
Comment (cloudvenger) [reply src/contracts/project.cairo:117]: This was a changed made by OpenZepplin. The `mint` without check does not exist anymore in the ERC-1155

**Review by tekkac — APPROVED** [review]
